io.cucumber.eclipse.python/src/io/cucumber/eclipse/python/launching/CucumberBehaveLaunchConfigurationDelegate.java

```diff
@@ -61,14 +79,70 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
 				.withVerbose(isVerbose)
 				.withNoCapture(isNoCapture)
 				.withDryRun(isDryRun);
+			
+			// Add message endpoint arguments to inject the formatter
+			List<String> additionalArgs = new ArrayList<>();
+			endpoint.addBehaveArguments(additionalArgs);
+			launcher.withArguments(additionalArgs);
+			
+			// Start the endpoint listener
+			endpoint.start();
+			launch.addProcess(endpoint);
 
-			Process process = launcher.launch();
+			// Launch the behave process with Python plugin path
+			Process process = launcher.launch(pythonPluginPath);
 			IProcess iProcess = DebugPlugin.newProcess(launch, process, "Cucumber Behave");
 			iProcess.setAttribute(IProcess.ATTR_PROCESS_TYPE, "cucumber.behave");
 		} catch (IOException e) {
+			if (endpoint != null) {
+				endpoint.terminate();
+			}
 			throw new CoreException(new Status(IStatus.ERROR, Activator.PLUGIN_ID, 
 				"Failed to launch behave process", e));
+		} catch (CoreException e) {
+			if (endpoint != null) {
+				endpoint.terminate();
+			}
+			throw e;
+		} catch (RuntimeException e) {
+			if (endpoint != null) {
+				endpoint.terminate();
+			}
+			throw e;
+		}
+	}
+	
+	/**
+	 * Get the path to the Python plugin directory
+	 * 
+	 * @return Path to python-plugins directory
+	 * @throws CoreException if path cannot be determined
+	 */
+	private String getPythonPluginPath() throws CoreException {
+		try {
+			Bundle bundle = Activator.getDefault().getBundle();
+			URL pluginURL = FileLocator.find(bundle, new Path("python-plugins"), null);
+			if (pluginURL != null) {
+				URL resolvedURL = FileLocator.resolve(pluginURL);
```

Comment:
We should better use org.eclipse.core.runtime.FileLocator.toFileURL(URL) and then pass it to new File(URL.toUri())